### PR TITLE
chore: remove csi-lib-iscsi vendor

### PR DESCRIPTION
**Pull request by andyzhangx**
**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
chore: remove csi-lib-iscsi vendor

there's a plan to move https://github.com/kubernetes-csi/csi-lib-iscsi to be part of https://github.com/kubernetes-csi/csi-driver-iscsi, csi-lib-iscsi repo  will be deprecated in the near future

This PR addresses this work item

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
cc @xing-yang @msau42 @humblec 

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

**Comment by humblec** [comment]
Thanks @andyzhangx .. just to mention there are some direct consumers of that library, so archival of that should go some cadence.. regardless getting it as part of this driver is a good first step!

**Comment by andyzhangx** [comment]
> Thanks @andyzhangx .. just to mention there are some direct consumers of that library, so archival of that should go some cadence.. regardless getting it as part of this driver is a good first step!

@humblec do you have the repo link of those direct consumers of this library?

**Comment by humblec** [comment]
> > Thanks @andyzhangx .. just to mention there are some direct consumers of that library, so archival of that should go some cadence.. regardless getting it as part of this driver is a good first step!
> 
> @humblec do you have the repo link of those direct consumers of this library?

https://github.com/Seagate is one of the big consumer of that library and making its way to their CSI driver.

**Comment by humblec** [comment]
may be a couple of commits here can be squashed.. rest looks fine to me . thanks !

**Comment by xing-yang** [comment]
It looks like https://github.com/Seagate forked the csi-lib-iscsi repo.

**Comment by xing-yang** [comment]
Looks good to me.  Please squash the commits.

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -1,48 +1,19 @@
...
-        run: golangci-lint run
-        if: ${{ failure() }}
+    go_lint:
+        name: Go Lint
+        runs-on: ubuntu-latest
+        steps:
+            - name: Set up Go 1.x
+              uses: actions/setup-go@v4
+              with:
+                go-version: ^1.19
```
[the remark this answers is not in the document]
Comment (andyzhangx) [reply .github/workflows/golangci-lint.yml:13]: that's minimum golang requirement, it's actully using 1.21 now:
```
Successfully set up Go version ^1.19
/opt/hostedtoolcache/go/1.21.6/x64/bin/go env GOMODCACHE
/opt/hostedtoolcache/go/1.21.6/x64/bin/go env GOCACHE
/home/runner/go/pkg/mod
/home/runner/.cache/go-build
Warning: Restore cache failed: Dependencies file is not found in /home/runner/work/csi-driver-iscsi/csi-driver-iscsi. Supported file pattern: go.sum
go version go1.21.6 linux/amd64
```

**Comment by andyzhangx** [comment]
> minor comment otherwise it looks good to me..

@humblec I have squashed to 2 commits now, 1 for lib movement and another for golint change commit, could you approve? thanks.

**Pull request merged**